### PR TITLE
Refactoring NNPTP21-1 project

### PR DESCRIPTION
**Pull request by dejv67**
Pull request contains improvements and code cleanup.

### CODE REVIEW

File: INPTPZ1/Mathematics/ComplexNumber.cs [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+    public class ComplexNumber
+    {
+
+        public readonly static ComplexNumber Zero = new ComplexNumber()
+        {
+            RealPart = 0,
+            ImaginaryPart = 0
+        };
+        public double RealPart { get; set; }
+        public float ImaginaryPart { get; set; }
```
Comment (rodi0878) [inline INPTPZ1/Mathematics/ComplexNumber.cs]: change real and imaginary parts to same data type

File: INPTPZ1/Mathematics/ComplexNumber.cs [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+        }
+
+        public override string ToString()
+        {
+            return $"({RealPart} + {ImaginaryPart}i)";
+        }
+
+        internal ComplexNumber Divide(ComplexNumber b)
+        {
+            ComplexNumber tmp = this.Multiply(new ComplexNumber() { RealPart = b.RealPart, ImaginaryPart = -b.ImaginaryPart });
```
Comment (rodi0878) [inline INPTPZ1/Mathematics/ComplexNumber.cs]: tmp?

File: INPTPZ1/Mathematics/ComplexNumber.cs [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+            var tmp2 = b.RealPart * b.RealPart + b.ImaginaryPart * b.ImaginaryPart;
```
Comment (rodi0878) [inline INPTPZ1/Mathematics/ComplexNumber.cs]: tmp2?

File: INPTPZ1/Mathematics/Polynomial .cs [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+        }
+
+        /// <summary>
+        /// ToString
+        /// </summary>
+        /// <returns>String representation of polynomial</returns>
+        public override string ToString()
+        {
+            string polynomAsString = "";
+            int i = 0;
```
Comment (rodi0878) [inline INPTPZ1/Mathematics/Polynomial .cs]: why here and not directly in for? also see "j"

File: INPTPZ1/NewtonFractal.cs [hunk 5]
```diff
@@ -0,0 +1,149 @@
...
+    /// See more at: https://en.wikipedia.org/wiki/Newton_fractal
+    /// </summary>
+    class NewtonFractal
+    {
+        private const double POWER = 2;
+        private const double POLYNOMIAL_QUOTIENT_TOLERANCE = 0.5;
+        private const double ROOT_TOLERANCE = 0.01;
+        private const int MAX_NEWTONS_ITERATIONS = 30;
+        private const double INITIAL_COORDINATES = 0.0001;
+        private const string DEFAUL_FILENAME = "../../../out.png";
```
Comment (rodi0878) [inline INPTPZ1/NewtonFractal.cs]: typo

File: INPTPZ1/NewtonFractal.cs [hunk 6]
```diff
@@ -0,0 +1,149 @@
...
+            ResultPictureWidth = int.Parse(arguments[0]);
+            ResultPictureHeight = int.Parse(arguments[1]);
+            XMin = double.Parse(arguments[2]);
+            XMax = double.Parse(arguments[3]);
+            YMin = double.Parse(arguments[4]);
+            YMax = double.Parse(arguments[5]);
+            FileName = arguments[6];
+        }
+
+        private static void CreatePicture()
```
Comment (rodi0878) [inline INPTPZ1/NewtonFractal.cs:57]: too complex method, make it shorter and simpler

File: INPTPZ1/NewtonFractal.cs [hunk 7]
```diff
@@ -0,0 +1,149 @@
...
+            polynomial.AddCoefficient(ComplexNumber.Zero);
+            polynomial.AddCoefficient(ComplexNumber.Zero);
+            polynomial.AddCoefficient(new ComplexNumber() { RealPart = 1 });
+            return polynomial;
+
+        }
+
+        private static float SolveEquation(ref ComplexNumber root, Polynomial polynomial, Polynomial polynomialDerivation)
+        {
+            float it = 0;
```
Comment (rodi0878) [inline INPTPZ1/NewtonFractal.cs]: why are iterations in float?

File: INPTPZ1/NewtonFractal.cs [hunk 8]
```diff
@@ -0,0 +1,149 @@
...
+                }
+                it++;
+            }
+            return it;
+        }
+
+        private static int FindRoots(ComplexNumber root, ref List<ComplexNumber> roots)
+        {
+            var known = false;
+            var rootId = 0;
```
Comment (rodi0878) [inline INPTPZ1/NewtonFractal.cs]: according to use this is index and not id

**Review by rodi0878 — CHANGES_REQUESTED** [review]
There is quite a lot of unclean things... Please look at attached comments and update your pull request.

**Comment by dejv67** [comment]
Required unclean parts of code from first pull request were fixed.

**Comment by rodi0878** [comment]
Accepted

**Pull request closed without merging**